keras_cv/layers/preprocessing/random_flip.py

```diff
@@ -162,50 +141,111 @@ def augment_bounding_boxes(
                 "Please specify a bounding box format in the constructor. i.e."
                 "`RandomFlip(bounding_box_format='xyxy')`"
             )
-        bounding_boxes = bounding_boxes.copy()
+        bounding_boxes = bounding_box.to_dense(bounding_boxes)
         bounding_boxes = bounding_box.convert_format(
             bounding_boxes,
             source=self.bounding_box_format,
             target="rel_xyxy",
-            images=image,
+            images=raw_images,
+        )
+        boxes = bounding_boxes["boxes"]
+        batch_size = tf.shape(boxes)[0]
+        max_boxes = tf.shape(boxes)[1]
+        flip_horizontals = transformations["flip_horizontals"]
+        flip_verticals = transformations["flip_verticals"]
+
+        # broadcast
+        flip_horizontals = (
+            tf.ones(shape=(batch_size, max_boxes, 4))
+            * flip_horizontals[:, tf.newaxis, :]
         )
-        bounding_boxes = tf.cond(
-            transformation["flip_horizontal"],
-            lambda: RandomFlip._flip_bounding_boxes_horizontal(bounding_boxes),
-            lambda: bounding_boxes,
+        flip_verticals = (
+            tf.ones(shape=(batch_size, max_boxes, 4))
+            * flip_verticals[:, tf.newaxis, :]
+        )
+
+        boxes = tf.where(
+            flip_horizontals > (1.0 - self.rate),
+            self._flip_boxes_horizontal(boxes),
+            boxes,
         )
-        bounding_boxes = tf.cond(
-            transformation["flip_vertical"],
-            lambda: RandomFlip._flip_bounding_boxes_vertical(bounding_boxes),
-            lambda: bounding_boxes,
+        boxes = tf.where(
+            flip_verticals > (1.0 - self.rate),
+            self._flip_boxes_vertical(boxes),
+            boxes,
         )
+
+        bounding_boxes = bounding_boxes.copy()
+        bounding_boxes["boxes"] = boxes
         bounding_boxes = bounding_box.clip_to_image(
             bounding_boxes,
             bounding_box_format="rel_xyxy",
-            images=image,
+            images=raw_images,
         )
         bounding_boxes = bounding_box.convert_format(
             bounding_boxes,
             source="rel_xyxy",
             target=self.bounding_box_format,
             dtype=self.compute_dtype,
-            images=image,
+            images=raw_images,
         )
-        return bounding_box.to_ragged(bounding_boxes)
+        return bounding_boxes
```

Comment:
do you not need the to_ragged() call here?

Reply:
VectorizedBaseImageAugmentationLayer calls to_ragged() after the augmentation, so it is dulplicated.

https://github.com/keras-team/keras-cv/blob/1aa709156e6736c692346dd6e2190bbc26db3402/keras_cv/layers/preprocessing/vectorized_base_image_augmentation_layer.py